pkg/apis/config/sigstore_keys.go

```diff
@@ -133,56 +87,171 @@ func NewSigstoreKeysFromConfigMap(config *corev1.ConfigMap) (*SigstoreKeysMap, e
+func ConvertSigstoreKeys(_ context.Context, source *v1alpha1.SigstoreKeys) *SigstoreKeys {
+	sk := &SigstoreKeys{}
+	sk.MediaType = "application/vnd.dev.sigstore.trustedroot+json;version=0.1"
+	sk.CertificateAuthorities = make([]*pbtrustroot.CertificateAuthority, len(source.CertificateAuthorities))
 	for i := range source.CertificateAuthorities {
 		sk.CertificateAuthorities[i] = ConvertCertificateAuthority(source.CertificateAuthorities[i])
 	}
 
-	sk.TLogs = make([]TransparencyLogInstance, len(source.TLogs))
+	sk.Tlogs = make([]*pbtrustroot.TransparencyLogInstance, len(source.TLogs))
 	for i := range source.TLogs {
-		sk.TLogs[i] = ConvertTransparencyLogInstance(source.TLogs[i])
+		sk.Tlogs[i] = ConvertTransparencyLogInstance(source.TLogs[i])
 	}
 
-	sk.CTLogs = make([]TransparencyLogInstance, len(source.CTLogs))
+	sk.Ctlogs = make([]*pbtrustroot.TransparencyLogInstance, len(source.CTLogs))
 	for i := range source.CTLogs {
-		sk.CTLogs[i] = ConvertTransparencyLogInstance(source.CTLogs[i])
+		sk.Ctlogs[i] = ConvertTransparencyLogInstance(source.CTLogs[i])
 	}
 
-	sk.TimeStampAuthorities = make([]CertificateAuthority, len(source.TimeStampAuthorities))
+	sk.TimestampAuthorities = make([]*pbtrustroot.CertificateAuthority, len(source.TimeStampAuthorities))
 	for i := range source.TimeStampAuthorities {
-		sk.TimeStampAuthorities[i] = ConvertCertificateAuthority(source.TimeStampAuthorities[i])
+		sk.TimestampAuthorities[i] = ConvertCertificateAuthority(source.TimeStampAuthorities[i])
 	}
+	return sk
 }
 
 // ConvertCertificateAuthority converts public into private CertificateAuthority
-func ConvertCertificateAuthority(source v1alpha1.CertificateAuthority) CertificateAuthority {
-	return CertificateAuthority{
-		Subject: DistinguishedName{
+func ConvertCertificateAuthority(source v1alpha1.CertificateAuthority) *pbtrustroot.CertificateAuthority {
+	return &pbtrustroot.CertificateAuthority{
+		Subject: &pbcommon.DistinguishedName{
 			Organization: source.Subject.Organization,
 			CommonName:   source.Subject.CommonName,
 		},
-		URI:       *source.URI.DeepCopy(),
-		CertChain: source.CertChain,
+		Uri:       source.URI.String(),
+		CertChain: DeserializeCertChain(source.CertChain),
+		ValidFor: &pbcommon.TimeRange{
+			Start: &timestamppb.Timestamp{
+				Seconds: 0, // TODO: Add support for time range to v1alpha1.CertificateAuthority
+			},
+		},
 	}
 }
 
 // ConvertTransparencyLogInstance converts public into private
 // TransparencyLogInstance.
-func ConvertTransparencyLogInstance(source v1alpha1.TransparencyLogInstance) TransparencyLogInstance {
-	return TransparencyLogInstance{
-		BaseURL:       *source.BaseURL.DeepCopy(),
-		HashAlgorithm: source.HashAlgorithm,
-		PublicKey:     source.PublicKey,
+func ConvertTransparencyLogInstance(source v1alpha1.TransparencyLogInstance) *pbtrustroot.TransparencyLogInstance {
+	pbpk, pk, err := DeserializePublicKey(source.PublicKey)
+	if err != nil {
+		return nil // TODO: log error? Add return error?
```

Comment:
Yeah, I think we should return the error instead of silently eating it. Here and in L151, since we return it from DeserializePublicKey below.